gatsby-node.js

```diff
@@ -194,60 +84,179 @@ async function doQuery({
     );
   }
 
-  setQueryStatus(`graphql resulted in ${objects.length} records`);
+  // return a map by id for later use
+  return objects.reduce((acc, object = {}) => {
```

Comment:
alternatively this could be `Object.fromEntries(objects.map(object => [object.objectID, object]`

Reply:
Good point. If you think that's more readable than the `reduce` I'm happy to update!